packages/i18next/tests/findSupportedLanguage.test.ts

```diff
@@ -0,0 +1,19 @@
+import { findSupportedLanguage } from "../src/i18nextPlugin.ts";
+
+test("when no user language match the supported languages, return undefined", () => {
+    const result = findSupportedLanguage(["fr-CA", "fr"], ["en-US", "en"]);
+
+    expect(result).toBeUndefined();
+});
+
+test("when a user language match a supported language, return the matching language", () => {
+    const result = findSupportedLanguage(["fr-CA", "fr"], ["en-US", "en", "fr"]);
+
+    expect(result).toBe("fr");
+});
+
+test("when multiple user language matches a supported language, return the left most user language", () => {
+    const result = findSupportedLanguage(["fr-CA", "en", "fr"], ["en-US", "en", "fr"]);
+
+    expect(result).toBe("en");
+});
```

Comment:
I think you're missing a test case.

Let's say I have in my preferred languages "fr" and "fr-CA", and supported languages are "fr-CA", "fr" and "en". I would expect it to match the most specific language, here being "fr-CA"

Reply:
Yeah indeed, I am not sure it should be this plugin job to this thought. Are you aware of any native API that would help with that? Is this something that you code in OV?

Reply:
Ok so that part of the case was indeed not very well done. Thank you for pointing this out.

I refactored the findSupportedPreferredLanguage to use [Intl.LocaleMatcher](https://github.com/tc39/proposal-intl-localematcher). Since the spec is currently in stage 1, I used a polyfill from [FormatJs](https://formatjs.io/docs/polyfills/intl-localematcher) thought.

I added several new test cases to make sure it works as expected.